pkg/persistentvolumeclaim/jiva.go

```diff
@@ -68,34 +70,66 @@ func DescribeJivaVolumeClaim(c *client.K8sClient, pvc *corev1.PersistentVolumeCl
 	// 4. Print the Portal Information
+	replicaPodIPAndModeMap := make(map[string]string)
+	jvStatus := ""
 	if jv != nil {
 		util.TemplatePrinter(volume.JivaPortalTemplate, jv)
+		// Create Replica IP to Mode Map
+		if jv.Status.ReplicaStatuses != nil && len(jv.Status.ReplicaStatuses) != 0 {
+			for _, replicaStatus := range jv.Status.ReplicaStatuses {
+				replicaPodIPAndModeMap[strings.Split(replicaStatus.Address, ":")[1][2:]] = replicaStatus.Mode
+			}
+		}
+		jvStatus = jv.Status.Status
+	} else {
+		fmt.Println()
 	}
 	// 5. Fetch the Jiva controller and replica pod details
-	podList, err := c.GetJVTargetPod(jv.Name)
+	podList, err := c.GetJVTargetPod(vol.Name)
 	if err == nil {
-		fmt.Printf("Controller and Replica Pod Details :" + "\n-----------------------------------\n")
+		fmt.Println("Controller and Replica Pod Details :")
+		fmt.Println("-----------------------------------")
 		var rows []metav1.TableRow
 		for _, pod := range podList.Items {
-			rows = append(rows, metav1.TableRow{Cells: []interface{}{
-				pod.Namespace, pod.Name,
-				util.GetReadyContainers(pod.Status.ContainerStatuses),
-				pod.Status.Phase, util.Duration(time.Since(pod.ObjectMeta.CreationTimestamp.Time)),
-				pod.Status.PodIP, pod.Spec.NodeName}})
+			if !strings.Contains(pod.Name, "-ctrl-") {
+				mode := ""
+				// If the IP doesnot exist, keep the mode as empty
+				if val, ok := replicaPodIPAndModeMap[pod.Status.PodIP]; ok {
+					mode = val
+				}
+				rows = append(rows, metav1.TableRow{Cells: []interface{}{
+					pod.Namespace, pod.Name, mode,
+					pod.Spec.NodeName, pod.Status.Phase, pod.Status.PodIP,
+					util.GetReadyContainers(pod.Status.ContainerStatuses),
+					util.Duration(time.Since(pod.ObjectMeta.CreationTimestamp.Time))}})
+			} else {
+				rows = append(rows, metav1.TableRow{Cells: []interface{}{
+					pod.Namespace, pod.Name, jvStatus,
+					pod.Spec.NodeName, pod.Status.Phase, pod.Status.PodIP,
+					util.GetReadyContainers(pod.Status.ContainerStatuses),
+					util.Duration(time.Since(pod.ObjectMeta.CreationTimestamp.Time))}})
+			}
 		}
-		util.TablePrinter(util.PodDetailsColumnDefinations, rows, printers.PrintOptions{Wide: true})
+		util.TablePrinter(util.JivaPodDetailsColumnDefinations, rows, printers.PrintOptions{Wide: true})
 	} else {
-		fmt.Printf("Controller and Replica Pod Details :\n-----------------------------------\nNo Controller and Replica pod exists for the JivaVolume\n")
+		fmt.Printf("Controller and Replica Pod Details :")
+		fmt.Println("-----------------------------------")
+		fmt.Println("No Controller and Replica pod exists for the JivaVolume")
 	}
 	// 6. Fetch the replica PVCs and create rows for cli-runtime
 	var rows []metav1.TableRow
-	pvcList, err := c.GetPVCs(jv.Namespace, nil, "openebs.io/component=jiva-replica,openebs.io/persistent-volume="+jv.Name)
+	pvcList, err := c.GetPVCs(c.Ns, nil, "openebs.io/component=jiva-replica,openebs.io/persistent-volume="+vol.Name)
```

Comment:
The replica PVCs aren't supposed to be present in the JIVA_NAMESPACE?, i.e. was this change intended?

Reply:
- The above caused a NPE when a JV goes missing. The JIVA_NAMESPACE gets assigned to the client before the method is called. Ref:-
https://github.com/openebs/openebsctl/blob/a5be7503ba4a326545073538980fdcc8fe24e51d/pkg/persistentvolumeclaim/persistentvolumeclaim.go#L52